Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp

```diff
@@ -279,27 +286,60 @@ namespace AzFramework
         return queue.m_delayed.empty() ? CommandQueueStatus::NoCommandsLeft : CommandQueueStatus::HasCommandsLeft;
     }
 
-    AZStd::pair<EntitySpawnTicket::Id, void*> SpawnableEntitiesManager::CreateTicket(AZ::Data::Asset<Spawnable>&& spawnable)
+    void* SpawnableEntitiesManager::CreateTicket(AZ::Data::Asset<Spawnable>&& spawnable)
     {
         static AZStd::atomic_uint32_t idCounter { 1 };
 
         auto result = aznew Ticket();
         result->m_spawnable = AZStd::move(spawnable);
+        result->m_ticketId = idCounter++;
+
+        m_totalTickets++;
+        m_ticketsPendingRegistration++;
+        AZ_Assert(
+            m_ticketsPendingRegistration <= m_totalTickets, "There are less total tickets than there are tickets pending registration.");
```

Comment:
nit: Add context to the error message ("ticket" is really generic, so even just adding that it's an issue related to Spawnables should help debug the issue if asserts are set to non-breaking)

Reply:
Added a bit of extra information to make it clear that it's about spawnables.